libcasr/src/ubsan.rs

```diff
@@ -72,65 +68,99 @@ impl Severity for UbsanWarning {
+                    )));
+                };
+                let file = cap.get(1).unwrap().as_str().to_string();
+                let line = cap.get(2).unwrap().as_str().parse::<u64>();
+                let Ok(line) = line else {
+                    return Err(Error::Casr(format!(
+                        "Couldn't parse crashline line: {crashline}"
+                    )));
+                };
+                if let Some(column) = cap.get(3) {
+                    let column = column.as_str().parse::<u64>();
+                    let Ok(column) = column else {
+                        return Err(Error::Casr(format!(
+                            "Couldn't parse crashline column: {crashline}"
+                        )));
+                    };
+                    return Ok(CrashLine::Source(DebugInfo { file, line, column }));
+                }
+                Ok(CrashLine::Source(DebugInfo {
+                    file,
+                    line,
+                    column: 0,
+                }))
+            }
+        }
+
         let message = self.ubsan_report();
         if message.is_empty() {
             return Err(Error::Casr("Empty ubsan message".to_string()));
         }
-        // If there is no stacktrace use crashline from first string
-        // May be not absolute
-        // Else use first string from stacktrace
-        let crashline = if let Some(crashline) = message
+        // If there is no stacktrace use first string from stacktrace
+        // Or combine with crashline from UbsanWarning first string
+        // Else use crashline from UbsanWarning first string
+        let header_crashline = get_crash_line(&message[0]);
+        if let Some(crashline) = message
             .iter()
             .skip(1)
             .find(|line| line.starts_with("    #0 "))
         {
-            crashline
-        } else {
-            &message[0]
-        };
-
-        if let Ok(crashline) = UbsanWarning::parse_stacktrace_entry(crashline) {
-            if !crashline.debug.file.is_empty() {
-                Ok(CrashLine::Source(crashline.debug))
-            } else if !crashline.module.is_empty() && crashline.offset != 0 {
-                Ok(CrashLine::Module {
-                    file: crashline.module,
-                    offset: crashline.offset,
-                })
+            let Ok(stack_crashline) = get_crash_line(crashline) else {
+                return header_crashline;
+            };
+            let Ok(header_crashline) = header_crashline else {
+                // Return error as it is
+                return header_crashline;
+            };
+            let CrashLine::Source(stack_crashline) = stack_crashline else {
+                // header_crashline may be CrashLine::Source
+                return Ok(header_crashline);
+            };
+            let CrashLine::Source(header_crashline) = header_crashline else {
+                return Ok(header_crashline);
```

Comment:
Why do we return `header_crashline`? We've parsed `stack_crashline` with sources.

Reply:
Done